test/sql/window/test_window_sort.test_slow

```diff
@@ -69,7 +69,9 @@ select * from rownum_window order by l_extendedprice
 ----
 
 query IIIII nosort q3
-select l_extendedprice, list_value(l_shipdate), list_value(l_commitdate), list_value(l_receiptdate), row_number() over (order by l_extendedprice) from lineitem_d;
+select l_extendedprice, list_value(l_shipdate), list_value(l_commitdate), list_value(l_receiptdate), row_number() over (order by l_extendedprice) 
+from lineitem_d
+order by l_extendedprice;
```

Comment:
I am a bit concerned with this, as it seems that the parallel scan of the window operator does not yield deterministic results. If someone executes this query, they would expect the output to be ordered by `l_extendedprice` as per the window specification and not have to add another `order by`.

We can have a deterministic output order for parallel scans using _batch indices_, which the window operator also needs. I have implemented this in the `PhysicalOrder` operator. You can check there for an example. There, the `batch_index` depends on the `block_idx` of the `PayloadScanner`.

The general idea of batch indices is that you assign an index to a range of values that are assigned to a thread. For example:

| thread | chunks | batch index |
|:--|:--|:--|
| 0 | 0-10 | 0 |
| 1 | 10-20 | 1 |
| 2 | 20-30 | 2 |
| 4 | 30-40 | 3 |

Thread 0 can keep outputting chunks using the same batch index of 0, and they will be appended in order to a `BatchedDataCollection` further in the pipeline if order matters there.

The number of chunks per batch is up to you. See what works best.

Reply:
Thanks - this is a good point. 

The big question for me is how to handle work stealing. Would it make sense to assign a batch index to each work block (returned by `GetScanner`) and set the order up to be (partition, block)?

Reply:
I tried this and the problem is that work stealing results in threads potentially working on "earlier" batch indexes that they steal from skewed partitions.

Reply:
Talked to @Mytherin and we agreed that it is only reasonable to expected ordered output when there is a single partition. Got that working, but anything more complex will need to wait until the accelerator (e.g., segment tree) construction has been parallelised.